noxfile.py

```diff
@@ -59,3 +64,151 @@ def docs(session):
             "docs",  # source directory
             "docs/_build/" + dest,  # output directory
         )
+
+
+@nox.session
+def release(session):
+    package_name = "packaging"
+    version_file = Path(f"{package_name}/__about__.py")
+
+    try:
+        release_version = _get_version_from_arguments(session.posargs)
+    except ValueError as e:
+        session.error(f"Invalid arguments: {e}")
+
+    # Check state of working directory and git
+    _check_working_directory_state(session)
+    _check_git_state(session, release_version)
+
+    # Bump to release version
+    _bump(session, version=release_version, file=version_file, kind="release")
+
+    # Tag the release commit
+    session.run("git", "tag", "-s", release_version, external=True)
+
+    # Bump for development
+    major, minor = map(int, release_version.split("."))
+    next_version = f"{major}.{minor + 1}.dev0"
+    _bump(session, version=next_version, file=version_file, kind="development")
+
+    # Checkout the git tag
+    session.run("git", "checkout", "-q", release_version, external=True)
+
+    # Build the distribution
+    session.run("python", "setup.py", "sdist", "bdist_wheel")
+
+    # Check what files are in dist/ for upload.
+    files = glob.glob(f"dist/*")
+    assert sorted(files) == [
+        f"dist/{package_name}-{release_version}.tag.gz",
+        f"dist/{package_name}-{release_version}-py2.py3-none-any.whl",
+    ], f"Got the wrong files: {files}"
+
+    # Get back out into master
+    session.run("git", "checkout", "-q", "master", external=True)
```

Comment:
Is this just to prevent accidental mistakes in the future?

Reply:
Sorta -- it serves to create "start on master, end on master" workflow.

I'd found that being switched to a tag after cutting the release is odd and to be on the branch you started on initially is much nicer. :)